src/main/java/com/powsybl/openloadflow/OpenLoadFlowProvider.java

```diff
@@ -104,7 +104,8 @@ private void updateAcState(Network network, LoadFlowParameters parameters, OpenL
         }
         try {
             // update network state
-            if (atLeastOneComponentHasToBeUpdated || parametersExt.isAlwaysUpdateNetwork()) {
+            if (atLeastOneComponentHasToBeUpdated && result.isSuccess()
+                    || parametersExt.isAlwaysUpdateNetwork()) {
```

Comment:
Just to mention that `alwaysUpdateNetwork` parameter makes the network to be updated even on `NO_CALCULATION` components, but that should be the expected behaviour.